build/root/Makefile.generated_files

```diff
@@ -37,81 +37,54 @@ SHELL := /bin/bash
-#
-#     make nonexistent
-#     # expect "No rule to make target"
-#     # verify `find .make/ -type f | wc -l` has many files
-#
-#     make gen_deepcopy
-#     # expect deepcopy-gen is built exactly once
-#     # expect many files to be regenerated
-#     # verify `find . -name zz_generated.deepcopy.go | wc -l` has files
-#     make gen_deepcopy
-#     # expect nothing to be rebuilt, finish in O(seconds)
-#     touch pkg/api/types.go
-#     make gen_deepcopy
-#     # expect one file to be regenerated
-#     make gen_deepcopy
-#     # expect nothing to be rebuilt, finish in O(seconds)
-#     touch vendor/k8s.io/code-generator/cmd/deepcopy-gen/main.go
-#     make gen_deepcopy
-#     # expect deepcopy-gen is built exactly once
-#     # expect many files to be regenerated
-#     # verify `find . -name zz_generated.deepcopy.go | wc -l` has files
-#     make gen_deepcopy
-#     # expect nothing to be rebuilt, finish in O(seconds)
-#
-#     make gen_conversion
-#     # expect conversion-gen is built exactly once
-#     # expect many files to be regenerated
-#     # verify `find . -name zz_generated.conversion.go | wc -l` has files
-#     make gen_conversion
-#     # expect nothing to be rebuilt, finish in O(seconds)
-#     touch pkg/api/types.go
-#     make gen_conversion
-#     # expect one file to be regenerated
-#     make gen_conversion
-#     # expect nothing to be rebuilt, finish in O(seconds)
-#     touch vendor/k8s.io/code-generator/cmd/conversion-gen/main.go
-#     make gen_conversion
-#     # expect conversion-gen is built exactly once
-#     # expect many files to be regenerated
-#     # verify `find . -name zz_generated.conversion.go | wc -l` has files
-#     make gen_conversion
-#     # expect nothing to be rebuilt, finish in O(seconds)
-#
-#     make all
-#     # expect it to build
-#
-#     make test
-#     # expect it to pass
-#
-#     make clean
-#     # verify `find . -name zz_generated.deepcopy.go | wc -l` is 0
-#     # verify `find . -name .make | wc -l` is 0
-#
-#     make all WHAT=cmd/kube-proxy
-#     # expect it to build
-#
-#     make clean
-#     make test WHAT=cmd/kube-proxy
-#     # expect it to pass
+#
+# Helper logic to calculate Go's dependency DAG ourselves.
+#
+
+# This is a file that will be emitted by the go2make tool, containing a
+# variable for each Go package in the project (including deps) which lists all
+# of the transitive deps of that package.  Each variable is named the same as
+# the package - for example the variable for `k8s.io/kubernetes/pkg/api` is
+# $(k8s.io/kubernetes/pkg/api).  This is roughly the same DAG that the Go
+# compiler uses.  These variables can be used to figure out if, for example,
+# generated code needs to be regerated.
+GO_PKGDEPS_FILE = go-pkgdeps.mk
+
+# Include the Go package dependencies file.  This will cause the rule of
+# the same name to be considered and if it is updated, make will restart and
+# reload the updated deps.
+sinclude $(META_DIR)/$(GO_PKGDEPS_FILE)
```

Comment:
is this including the `$(GO_PKGDEPS_FILE)` before it's updated? or is `make` smart enough to generate this file first, since there's a rule for it?

Reply:
Make will try to regenerate it before including it.